Userland/Libraries/LibWasm/AbstractMachine/AbstractMachine.cpp

```diff
@@ -155,7 +156,60 @@ InstantiationResult AbstractMachine::instantiate(Module const& module, Vector<Ex
     Vector<Vector<Reference>> elements;
     ModuleInstance auxiliary_instance;
 
-    // FIXME: Check that imports/extern match
+    module.for_each_section_of_type<ImportSection>([&](ImportSection const& section) {
+        for (auto [i, import_] : enumerate(section.imports())) {
+            auto extern_ = externs.at(i);
+            auto is_valid = import_.description().visit(
+                [&](MemoryType const& mem_type) -> bool {
+                    if (!extern_.has<MemoryAddress>())
+                        return false;
+                    auto other_mem_type = m_store.get(extern_.get<MemoryAddress>())->type();
+                    return other_mem_type.limits().matches(mem_type.limits());
+                },
+                [&](TableType const& table_type) -> bool {
+                    if (!extern_.has<TableAddress>())
+                        return false;
+                    auto other_table_type = m_store.get(extern_.get<TableAddress>())->type();
+                    return table_type.element_type() == other_table_type.element_type()
+                        && other_table_type.limits().matches(table_type.limits());
+                },
+                [&](GlobalType const& global_type) -> bool {
+                    if (!extern_.has<GlobalAddress>())
+                        return false;
+                    auto other_global_type = m_store.get(extern_.get<GlobalAddress>())->type();
+                    return global_type.type() == other_global_type.type()
+                        && global_type.is_mutable() == other_global_type.is_mutable();
+                },
+                [&](FunctionType const& type) -> bool {
+                    if (!extern_.has<FunctionAddress>())
+                        return false;
+                    auto other_type = m_store
+                                          .get(extern_.get<FunctionAddress>())
+                                          ->visit(
+                                              [&](WasmFunction const& wasm_func) { return wasm_func.type(); },
+                                              [&](HostFunction const& host_func) { return host_func.type(); });
```

Comment:
How did you get clang-format to do _this_?
pls
```diff
diff --git a/Userland/Libraries/LibWasm/AbstractMachine/AbstractMachine.cpp b/Userland/Libraries/LibWasm/AbstractMachine/AbstractMachine.cpp
index d23664b3a8..7c3dd3af2d 100644
--- a/Userland/Libraries/LibWasm/AbstractMachine/AbstractMachine.cpp
+++ b/Userland/Libraries/LibWasm/AbstractMachine/AbstractMachine.cpp
@@ -183,22 +183,14 @@ InstantiationResult AbstractMachine::instantiate(Module const& module, Vector<Ex
                 [&](FunctionType const& type) -> bool {
                     if (!extern_.has<FunctionAddress>())
                         return false;
-                    auto other_type = m_store
-                                          .get(extern_.get<FunctionAddress>())
-                                          ->visit(
-                                              [&](WasmFunction const& wasm_func) { return wasm_func.type(); },
-                                              [&](HostFunction const& host_func) { return host_func.type(); });
+                    auto other_type = m_store.get(extern_.get<FunctionAddress>())->visit([&](WasmFunction const& wasm_func) { return wasm_func.type(); }, [&](HostFunction const& host_func) { return host_func.type(); });
                     return type.results() == other_type.results()
                         && type.parameters() == other_type.parameters();
                 },
                 [&](TypeIndex type_index) -> bool {
                     if (!extern_.has<FunctionAddress>())
                         return false;
-                    auto other_type = m_store
-                                          .get(extern_.get<FunctionAddress>())
-                                          ->visit(
-                                              [&](WasmFunction const& wasm_func) { return wasm_func.type(); },
-                                              [&](HostFunction const& host_func) { return host_func.type(); });
+                    auto other_type = m_store.get(extern_.get<FunctionAddress>())->visit([&](WasmFunction const& wasm_func) { return wasm_func.type(); }, [&](HostFunction const& host_func) { return host_func.type(); });
                     auto& type = module.type(type_index);
                     return type.results() == other_type.results()
                         && type.parameters() == other_type.parameters();
```

Reply:
Uhh... I think I just vaguely suggested it by splitting it up on different lines and it just worked. Not really sure to be honest.